client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -243,12 +247,12 @@ const Checkout = React.createClass( {
 		}
 
 		return this.props.selectedFeature && isValidFeatureKey( this.props.selectedFeature )
-			? `/checkout/thank-you/features/${ this.props.selectedFeature }/${ this.props.sites.getSelectedSite().slug }/${ receiptId }`
-			: `/checkout/thank-you/${ this.props.sites.getSelectedSite().slug }/${ receiptId }`;
+			? `/checkout/thank-you/features/${ this.props.selectedFeature }/${ selectedSite.slug }/${ receiptId }`
+			: `/checkout/thank-you/${ selectedSite.slug }/${ receiptId }`;
```

Comment:
The same goes here, there's no handling for cases where no selectedSite is available but I'm not so sure what would be a sensible way to handle this :/
Other than these 2 urls I think I have changes that address the other points :)